homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -326,20 +334,25 @@ async def async_refresh_data(self, now):
         self.request_in_progress = False
         async_dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
 
-    @staticmethod
-    def _extract_start_date(battery_info):
-        """Extract the server date from the battery response."""
-        try:
-            return battery_info.answer["BatteryStatusRecords"]["OperationDateAndTime"]
-        except KeyError:
-            return None
-
     async def async_get_battery(self):
         """Request battery update from Nissan servers."""
 
         try:
             # Request battery update from the car
             _LOGGER.debug("Requesting battery update, %s", self.leaf.vin)
+            start_date = None
+            try:
+                start_server_info = await self.hass.async_add_executor_job(
+                    self.leaf.get_latest_battery_status
+                )
+            except TypeError:  # pycarwings2 can fail if Nissan returns nothing
+                _LOGGER.debug("Battery status check returned nothing")
+            else:
+                if not start_server_info:
+                    _LOGGER.debug("Battery status check failed")
+                else:
+                    start_date = _extract_start_date(start_server_info)
+            await asyncio.sleep(1)  # Critical sleep
             request = await self.hass.async_add_executor_job(self.leaf.request_update)
```

Comment:
This isn't polling too frequently so if we stored the last value, we could request the update and then schedule a check 60 seconds later to get the new data and check to see if its changed

Reply:
Can you explain a bit more what you mean? Is not polling too frequently a good thing?

I am open to scheduling checks and waiting for them in other methods, but my intention here is to make the code work without diverging too much from @filcole 's creation.